Tasks/XamariniOS/xamarinios.ts

```diff
@@ -1,155 +1,133 @@
-var device = (buildForSimulator) ? 'iPhoneSimulator' : 'iPhone';
-tl.debug('device: ' + device);
-var xbuildLocation = tl.getInput('mdtoolLocation', false);
-var cwd = tl.getInput('cwd');
-
-// Get path to xbuild
-var xbuildToolPath = undefined;
-if (xbuildLocation) {
-    xbuildToolPath = path.join(xbuildLocation, 'xbuild');
-    tl.checkPath(xbuildToolPath, 'xbuild');
-} else {
-    xbuildToolPath = tl.which('xbuild', true);
-}
+import sign = require('ios-signing-common/ios-signing-common');
 
-// Find location of nuget
-var nugetPath = tl.which('nuget', true);
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
 
-//Process working directory
-var cwd = cwd || tl.getVariable('build.sourceDirectory') || tl.getVariable('build.sourcesDirectory');
-tl.cd(cwd);
+async function run() {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
 
-// Prepare function for tool execution failure
-var onFailedExecution = function (err) {
-    // Error executing
-    tl.debug('ToolRunner execution failure: ' + err);
-    tl.error('See https://go.microsoft.com/fwlink/?LinkId=760847');
-    tl.exit(1);
-}
-var deleteKeychain:trm.ToolRunner = null;
-var deleteProvProfile:trm.ToolRunner = null;
-var provProfileUUID = null;
-var signIdentity = null;
-
-function iosIdentity() {
-
-    var input = {
-        cwd: cwd,
-        unlockDefaultKeychain: tl.getBoolInput('unlockDefaultKeychain', false),
-        defaultKeychainPassword: tl.getInput('defaultKeychainPassword', false),
-        p12: tl.getPathInput('p12', false, false),
-        p12pwd: tl.getInput('p12pwd', false),
-        iosSigningIdentity: tl.getInput('iosSigningIdentity', false)
-    }
+        // Get build inputs
+        var solutionPath = tl.getPathInput('solution', true, true);
+        var configuration = tl.getInput('configuration', true);
+        var args = tl.getInput('args');
+        var packageApp = tl.getBoolInput('packageApp');
+        var buildForSimulator = tl.getBoolInput('forSimulator');
+        var device = (buildForSimulator) ? 'iPhoneSimulator' : 'iPhone';
+        tl.debug('device: ' + device);
+        var xbuildLocation = tl.getInput('mdtoolLocation', false);
+        var cwd = tl.getInput('cwd');
+
+        // Get path to xbuild
+        var xbuildToolPath = undefined;
+        if (xbuildLocation) {
+            xbuildToolPath = path.join(xbuildLocation, 'xbuild');
+            tl.checkPath(xbuildToolPath, 'xbuild');
+        } else {
+            xbuildToolPath = tl.which('xbuild', true);
+        }
+
+        // Find location of nuget
+        var nugetPath = tl.which('nuget', true);
+
+        // Restore NuGet packages of the solution
+        var nugetRunner = tl.tool(nugetPath);
+        nugetRunner.arg(['restore', solutionPath]);
+        await nugetRunner.exec();
+
+        //Process working directory
+        var cwd = cwd || tl.getVariable('build.sourceDirectory') || tl.getVariable('build.sourcesDirectory');
```

Comment:
Yes it was there for the very old agent, I want to basically remove build.sourcesDirectory since the task might be running in a release definition too. But will do that when porting tests to new framework.